cpp/src/arrow/filesystem/s3fs.h

```diff
@@ -102,6 +102,8 @@ struct ARROW_EXPORT S3Options {
   /// the region (environment variables, configuration profile, EC2 metadata
   /// server).
   std::string region;
+  long request_timeout_ms = -1;
+  long connect_timeout_ms = -1;
```

Comment:
We would probably want to be consistent accross the codebase and expose it as seconds, so perhaps:
```c++
  /// \brief Request timeout in seconds
  ///
  /// If not specified, the AWS SDK's default value will be used.
  util::optional<double> request_timeout;
```

Reply:
Changed it to double

Reply:
@pitrou So changing it to double actually is a bit problematic because the underlying AWS config data types are long. We'd have to convert them back to long if we use double here. Please let me know what you'd prefer

Reply:
Sorry for the delay. We want to make the API easy to use and passing seconds is the most natural choice. So we'll have to live with a `double`.

Reply:
addressed this